earth_enterprise/src/common/config/geConfigUtil.cpp

```diff
@@ -41,41 +41,14 @@ bool IsRedHat(void) {
 }
 
 uint GetNumCPUs(void) {
-  uint numcpus = 0;
-  std::ifstream in("/proc/cpuinfo");
-  if (in) {
-    std::string line;
-    while (std::getline(in, line)) {
-      if (StartsWith(line, "processor")) {
-        ++numcpus;
-      }
-    }
-  }
-
-  return std::max(uint(1), numcpus);
+  return std::max(uint(1), uint(sysconf(_SC_NPROCESSORS_ONLN)));
 }
 
+// want memory size in Kb
 uint64 GetPhysicalMemorySize(void) {
-  std::ifstream in("/proc/meminfo");
-  const std::string delimiters(" \t");
-  if (in) {
-    std::string line;
-    while (std::getline(in, line)) {
-      // will be a line of the form: MemTotal:       255944 kB
-      if (StartsWith(line, "MemTotal")) {
-        // parse out the kB.
-        std::vector<std::string> tokens;
-        TokenizeString(line, tokens, delimiters);
-        if (tokens.size() == 3 && tokens[2] == "kB") {
-          return strtouq(tokens[1].c_str(), NULL, 10) * 1000;
-        }
-        return 0;  // parse failure...too many tokens or not "kB"
-      }
-    }
-    return 0;  // parse failure
-  }
-
-  return 0;  // parse failure
+    uint64 pages = sysconf(_SC_PHYS_PAGES),
+           page_size = sysconf(_SC_PAGE_SIZE)/1024;
+    return (pages * page_size * 1000);
```

Comment:
Do we really need to return an incorrect memory size?

I think this should return the memory size in bytes, and any functions that use the memory size should make the proper calculations.

Reply:
This would necessitate a review to see where this function is called

Reply:
Yeah, this is the only place it is used:

```
src/fusion/gecombineterrain/gecombineterrain.cpp:290
```

Reply:
Do we want to go ahead and fix this here or open a new issue?

Reply:
I don't think we need a new issue.  I think `gecombineterrain` will work just fine using the actual amount of memory.